src/Angor/Client/Pages/Spend.razor

```diff
@@ -628,20 +644,39 @@
     private void Expand(int stageId)
     {
         expandedStageId = expandedStageId == stageId ? null : stageId;
+        selectedUtxos.Clear();
     }
 
-    private void HandleCheckboxChange(string trxId)
+    private void HandleCheckboxChange(string trxId, int outputIndex)
     {
+        var key = new UtxoKey(trxId, outputIndex);
 
-        if (selected.ContainsKey(trxId))
+        if (selectedUtxos.ContainsKey(key))
         {
-            selected.Remove(trxId);
+            selectedUtxos.Remove(key);
         }
         else
         {
-            selected.Add(trxId, null);
+            selectedUtxos[key] = true;
```

Comment:
This will always be true, we wont have a case that the value is false right?

Reply:
removing the key means it no longer exists in the dictionary, so there isn't a false value associated with it?
if so then i think you're right